src/extension/index.ts

```diff
@@ -67,19 +68,34 @@ export async function activate(context: ExtensionContext) {
         update();
     }
 
+    // For now, we keep file watch functionality limited to the API.
+    // After we're sure it's working well, consider also enabling it for files opened manually.
+    const watcher = watch([], {
+        ignoreInitial: true
+    }).on('change', async path => {
+        store.logs.removeFirst(log => log._uri === Uri.file(path).toString());
+        store.logs.push(...await loadLogs([Uri.file(path)]));
+    }).on('unlink', path => {
+        store.logs.removeFirst(log => log._uri === Uri.file(path).toString());
+    });
+    disposables.push(new Disposable(async () => await watcher.close()));
+
     // API
     const api = {
         async openLogs(logs: Uri[], _options: unknown, cancellationToken?: CancellationToken) {
+            watcher.add(logs.map(log => log.fsPath));
```

Comment:
Should `watcher.add` happen after `store.logs.push`?

Reply:
Hmm. I think that makes sense in case the following happens:
1. We open a log
2. The log is added to the watcher
3. For whatever reason, loadLogs takes forever to finish
4. The log is changed on the filesystem after the watcher starts watching the file but before loadLogs finishes
5. Race condition?

Wouldn't have thought about this otherwise, thanks for pointing it out :)

Reply:
Actually, turns out that loadLogs isn't even async, so we don't need to worry about that!
...which is a good thing, because my comment made me realize the same race conditions applied to multiple "change" events happening in short succession.

Reply:
Yes, `loadLogs` is labeled as `async` but doesn't actually `await` anything. In the past we used to await VS Code prompts.

However, if two "change" events come in short succession, it is still possible to race, correct? There's nothing preventing two `loadLogs` calls running at the same time.

Reply:
Right. Let me think of that and come back to you with a few ideas.

Reply:
Since we discussed don't anticipate short succession changes, I'd say a code comment acknowledging the possibility of a race is sufficient.

Reply:
Sounds good, comment added. Thinking about it - I think if loadLogs is sync, the chance the race condition will actually occur will be extremely small if not impossible - you're banking on another change event getting fired _after_ the removal but _before_ the push.
I'm not familiar with JavaScript multithreading as I'd like to be, but if the extension can only process one "change" event at a time like I suspect - that makes it race-free.